gym_bridges/envs/bridges_env.py

```diff
@@ -261,11 +268,96 @@ def reset(self, state=None, gap_count=None):
+                    y_window_coordinate,
+                    block_size,
+                    block_size,
+                )
+                if (
+                    self._state[y_coordinate][x_coordinate]
+                    == BridgesEnv.StateType.GROUND
+                ):
+                    screen.blit(textures[BridgesEnv.StateType.GROUND], rect)
+                elif (
+                    self._state[y_coordinate][x_coordinate]
+                    == BridgesEnv.StateType.BRICK
+                ):
+                    screen.blit(textures[BridgesEnv.StateType.BRICK], rect)
+                elif (
+                    self._state[y_coordinate][x_coordinate]
+                    == BridgesEnv.StateType.EMPTY
+                ):
+                    screen.blit(textures[BridgesEnv.StateType.EMPTY], rect)
+
     def render(self, mode="human"):
-        mapping = {
-            BridgesEnv.StateType.GROUND: "@@",
-            BridgesEnv.StateType.BRICK: "[]",
-            BridgesEnv.StateType.EMPTY: "  ",
-        }
-        flat_repr = tuple([mapping[x] for x in self._state.flatten()])
-        print((("%s" * self.shape[1] + "\n") * self.shape[0]) % flat_repr)
+        if mode == "human":
+            mapping = {
+                BridgesEnv.StateType.GROUND: "@@",
+                BridgesEnv.StateType.BRICK: "[]",
+                BridgesEnv.StateType.EMPTY: "  ",
+            }
+            flat_repr = tuple([mapping[x] for x in self._state.flatten()])
+            print((("%s" * self.shape[1] + "\n") * self.shape[0]) % flat_repr)
+        elif mode == "pygame":
+            if not self._pygame_initialized:
+                pygame.init()
+
+                self.screen = pygame.display.set_mode(
+                    (renderer_config.WINDOW_WIDTH, renderer_config.WINDOW_HEIGHT)
+                )
+                self.screen.fill(renderer_config.BLACK)
+
+                # Load the textures with a path relative to the game source code.
+                base_path = os.path.dirname(os.path.dirname(__file__))
+                textures_path = os.path.join(base_path, "renderer", "assets")
+
+                self.textures = {
+                    BridgesEnv.StateType.GROUND: pygame.image.load(
+                        os.path.join(textures_path, "grass_block.png")
+                    ),
+                    BridgesEnv.StateType.BRICK: pygame.image.load(
+                        os.path.join(textures_path, "mossy_stone_bricks.png")
+                    ),
+                    BridgesEnv.StateType.EMPTY: pygame.image.load(
+                        os.path.join(textures_path, "light_blue_wool.png")
+                    ),
+                }
+
+                self._pygame_initialized = True
+
+            self.draw_state(screen=self.screen, textures=self.textures)
+
+            # Drawing the state takes some time.
+            time.sleep(1)
+            for event in pygame.event.get():
+                if event.type == pygame.QUIT:
```

Comment:
I guess the idea here is that if the user closes the window, they are done with their program run? I wonder if that's a bit aggressive vs just closing the window and staying closed.

Reply:
I was following[ the convention here](https://stackoverflow.com/questions/19882415/closing-pygame-window).

Reply:
SGTM